packages/suite/src/components/wallet/AccountAnnouncement/index.tsx

```diff
@@ -18,6 +21,11 @@ interface AccountAnnouncementProps {
 
 export const AccountAnnouncement = ({ account }: AccountAnnouncementProps) => {
     const notifications = [];
+    const isCoinjoinBlockedByTor = useSelector(selectIsCoinjoinBlockedByTor);
+
+    if (isCoinjoinBlockedByTor) {
```

Comment:
i would move this condition into `TorDisconnected` component and return null if false from there,
imho it's better to have all the logic in one file

Reply:
or maybe @marekrjpolak wants to say something here, you are doing "account is not synced" in the same place right?

Reply:
- I more or less agree that logic hidden in the component looks tidier, on the other hand like this it's more consistent with the other announcements below, so no strong opinion here
- it's a bit inconsistent that the other announcements are rendered based on the `account` prop but `<TorDisconnected />` based on `selectedAccount` reducer instead. But they should generally be the same thing.
- I'd prefer to move `<TorDisconnected />` below the account null check and add it to `notifications` list like the other announcements so it's possible to display it simultaneously with something else (e.g. out of sync)

Reply:
I put `<TorDisconnected />` above the account null check because it happens some times that account is null and it maybe should not.
When there are exceptions in discovery `selectedAccount` returns without account `packages/suite/src/actions/wallet/selectedAccountActions.ts`

`selectedAccount` is not consistently return attributes but I guess there is a reason for that.

Reply:
Oh I see. I guess that's the difference between `<AccountMode />` which shows `warning` notifications based on `selectedAccount.mode` (as there is no `account` when discovery fails) and `<AccountAnnouncement />` which (until now) shows `info` notifications based on `selectedAccount.account`.

Anyway, `<AccountAnnouncement />` is always passed `selectedAccount.account` as the `account` prop so the 'nullishness' should be the same, no?

Reply:
Even if `selectedAccount.account` is always passed as `account`. I happens that `account` is sometimes nullish.
And there is at least 1 situation that it requires to display the Tor banner and `account` is undefined. 

If you start discovery of coinjoin account (with `isDevEnv` set to false in the code packages/suite-desktop/src/modules/request-interceptor.ts) and during discovery disable Tor, and go back to your coinjoin account, then account in `<AccountAnnouncement />` is undefined.

This is because when there is exeption `'account-not-loaded'` in `packages/suite/src/actions/wallet/selectedAccountActions.ts`  account is not passed and therefore it is undefined.


It might have some implications in the other announcements that are never going to be display when there is that exeption.

Reply:
Ok, if you believe that when the Tor banner is displayed, no other announcement has to be displayed as well (the longer I think about it the more correct it sounds), feel free to leave it as it is. :+1: